driver/ni/analog_read.cpp

```diff
@@ -7,254 +7,241 @@
-        return -1;
-    }
-    } else{
-        LOG(INFO) << "[NI Reader] configuring special timing for task " << this->reader_config.task_name;
+            "",
+            this->reader_config.sample_rate,
+            DAQmx_Val_Rising,
+            DAQmx_Val_ContSamps,
+            this->reader_config.sample_rate))) {
+            LOG(ERROR) << "[NI Reader] failed while configuring timing for task " <<
+                    this->reader_config.task_name;
+            this->ok_state = false;
+            return -1;
+        }
+    } else {
         if (this->checkNIError(ni::NiDAQmxInterface::CfgSampClkTiming(this->task_handle,
-                                                                    this->reader_config.timing_source.c_str(),
-                                                                    this->reader_config.sample_rate,
-                                                                    DAQmx_Val_Rising,
-                                                                    DAQmx_Val_ContSamps,
-                                                                    this->reader_config.sample_rate))){
-            LOG(ERROR) << "[NI Reader] failed while configuring timing for task " << this->reader_config.task_name;
+            this->reader_config.timing_source.c_str(),
+            this->reader_config.sample_rate,
+            DAQmx_Val_Rising,
+            DAQmx_Val_ContSamps,
+            this->reader_config.sample_rate))) {
+            LOG(ERROR) << "[NI Reader] failed while configuring timing for task " <<
+                    this->reader_config.task_name;
             this->ok_state = false;
             return -1;
         }
     }
-    this->numSamplesPerChannel = std::floor(this->reader_config.sample_rate / this->reader_config.stream_rate);
+    // we read data in chunks of numSamplesPerChannel such that we can send frames of data of size numSamplesPerChannel at the stream rate
+    // e.g. if we have 4 channels and we want to stream at 100Hz at a 1000hz sample rate
+    // make a make a call to read 10 samples at 100hz
+    this->numSamplesPerChannel = std::floor(
+        this->reader_config.sample_rate / this->reader_config.stream_rate);
     this->bufferSize = this->numAIChannels * this->numSamplesPerChannel;
     return 0;
-    // }r
 }
 
-
-void ni::AnalogReadSource::deleteScales(){
-    for(auto &channel : this->reader_config.channels){
-        if(channel.custom_scale){
-            if(channel.scale_type == "polyScale"){
-                delete[] channel.scale->polynomial.forward_coeffs;
-                delete[] channel.scale->polynomial.reverse_coeffs;
-            } else if(channel.scale_type == "tableScale"){
-                delete[] channel.scale->table.prescaled;
-                delete[] channel.scale->table.scaled;
-            }
-        }
-    }
-}
-
-void ni::AnalogReadSource::acquireData(){
-    while(this->running){
+void ni::AnalogReadSource::acquireData() {
+    while (this->running) {
```

Comment:
Why aren't we using a breaker here?

Reply:
will handle in next pr dealing with stability improvements